server/backend/api/app/crud/crud_utils.py

```diff
@@ -100,3 +101,31 @@ def is_app_admin(
     if access_roles and admin_role in access_roles:
         return True
     return False
+
+
+def get_aws_target_env() -> AwsTargetEnv:
+    # target_env is assigned from gov's AWS platform, does not exist in local (None).
+    return os.environ.get("target_env")
+
+
+def is_on_aws_prod() -> bool:
+    return get_aws_target_env() == AwsTargetEnv.PROD
+
+
+def use_api_instance_by_app(application: models.FamApplication) -> ApiInstanceEnv:
```

Comment:
I was thinking if put these methods in config.py or utils, maybe better to be in the utils?

Reply:
Having it at utils is great!! I didn't think about that. Thanks.
Maybe `get_aws_target_env()` can be in config file.

Also I didn't think about passing 'application' instead of app_environment and then check for `application_name`, but I think it is better, I told you you have sharp eyes. Do you think it is reusable in all cases? Probably yes. I would change that on my end.

And, in `server/admin_management` you didn't do the same logic and didn't pass application. Maybe it can be consistent?

Reply:
Hi Ian, that's because it's impossible to get application information from role (no role) when we grant app admin. And I don't want to make another database call to grab application information by application_admin_id or application from request body. Cause we're pretty sure in the case of granting app admin, we'll use the prod url for idim apis. So if you check the similar method I had for admin management use_api_instance_by_app_env, it uses app_env, and in the case of app_env is None, that means it's FAM.

Reply:
No worry, I just noticed it is not consistent and didn't recall your case.
It's good, just I need to see how to reuse yours (should be no problem), I will just wait on yours since I am having strange error on my workspace.

Reply:
yeah, oh yeah, it should be fine. We verify forest client number when grant delegated admin, and we should have the role.application.app_environment information.